java/bogdana_maksiuta/src/com/zoolatech/lecture2/tasks/_2/Robot.java

```diff
@@ -0,0 +1,189 @@
+    }
+
+    public void move() {
+        switch (getDirection()) {
+            case "east" -> {
+                w++;
+                eastDirection();
+            }
+            case "west" -> {
+                w--;
+                westDirection();
+            }
+            case "north" -> {
+                h--;
+                northDirection();
+            }
+            case "south" -> {
+                h++;
+                southDirection();
+            }
+        }
+    }
+
+    public void move(int tiles) {
+        switch (getDirection()) {
+            case "east" -> {
+                if ((getW() + tiles) < getWidthRoom()) {
+                    w += tiles;
+                    eastDirection();
+                } else {
+                    int allowed = getWidthRoom() - getW() - 1;
+                    System.out.println("There is " + allowed + "allowed");
+                }
+            }
+            case "west" -> {
+                if ((getW() - tiles) >= 0) {
+                    w -= tiles;
+                    westDirection();
+                } else {
+                    System.out.println("There is " + (getW()) + "allowed");
+                }
+
+            }
+            case "north" -> {
+                if ((getH() - tiles) >= 0) {
+                    h -= tiles;
+                    northDirection();
+                } else {
+                    System.out.println("There is " + getH() + "allowed");
+                }
+            }
+            case "south" -> {
+                if ((getH() + tiles) < getHeightRoom()) {
+                    h += tiles;
+                    southDirection();
+                } else {
+                    int allowed = getHeightRoom() - getH() - 1;
+                    System.out.println("There is " + allowed + "allowed");
+                }
+            }
+        }
+    }
+
+    public void turnRight() {
+        switch (getDirection()) {
+            case "west" -> northDirection();
+            case "north" -> eastDirection();
+            case "east" -> southDirection();
+            case "south" -> westDirection();
+        }
+    }
+
+    public void turnLeft() {
+        switch (getDirection()) {
+            case "west" -> southDirection();
+            case "south" -> eastDirection();
+            case "east" -> northDirection();
+            case "north" -> westDirection();
+        }
+    }
+
+    public void currentLocation() {
+        System.out.println("Your current position is: " + "[" + getW() + " : " + getH() + "] [weight : height]");
+    }
+
+    public void currentDirection() {
+        System.out.println("Direction: " + getDirection());
+    }
+
+    public void southDirection() {
+        if (h < (heightRoom - 1)) {
+            System.out.println("My direction: SOUTH");
+            setDirection("south");
+        } else if (h == (heightRoom - 1) && w < heightRoom / 2 && w >= 0) {
+            System.out.println("Hm. It's border. I need to go east");
```

Comment:
nice idea, but the task description was easier: you could just print that you stuck at the border